test/collection.js

```diff
@@ -506,7 +506,7 @@
     };
     collection.url = '/test';
     collection.fetch();
-    this.syncArgs.options.success();
+    this.syncArgs.options.success([]);
```

Comment:
I [preemptively return](https://github.com/jashkenas/backbone/pull/3663/files#diff-0d56d0d310de7ff18b3cef9c2f8f75dcR800) on `null` now, instead of letting it go through parse.